File: cad_automator/core_backend/processors/merger.py
import time
from ..interfaces import CadProcessor

class LayerMerger(CadProcessor):
    """
    [Concrete Product]
    具体的图层合并器实现。
    """
    def process(self, file_path: str, options: dict = None) -> dict:
        target_layer = options.get("target_layer", "0") if options else "0"
        source_pattern = options.get("source_pattern", "*")
        
        print(f"[Logic] 正在将符合 '{source_pattern}' 的图层合并到 '{target_layer}'")
        
        # 模拟耗时操作
        time.sleep(0.5)
        
        output_file = file_path.replace(".dxf", "_merged.dxf")
        
        return {
            "status": "success",
            "action": "layer_merge",
            "output_file": output_file,
            "details": f"Merged 5 layers into {target_layer}"
        }

File: cad_automator/rust_core/src/processors/merger.rs
use super::super::processor::{CadProcessor, ProcessResult};
use std::error::Error;
use std::path::PathBuf;
use serde_json::json;

// [Concrete Product 2]
// 图层合并处理器
pub struct LayerMerger;

impl CadProcessor for LayerMerger {
    fn name(&self) -> &str {
        "Layer Merger"
    }

    fn process(&self, file_path: &PathBuf, options: &serde_json::Value) -> Result<ProcessResult, Box<dyn Error>> {
        // 解析参数
        let target_layer = options["target_layer"].as_str().unwrap_or("0");
        
        println!("[Rust Core] 正在合并图层到: {}, 文件: {:?}", target_layer, file_path);

        // 实际逻辑:
        // let mut drawing = dxf::Drawing::load_file(file_path)?;
        // for entity in drawing.entities() { ... }
        
        let output = file_path.with_file_name(format!(
            "{}_merged.dxf",
            file_path.file_stem().unwrap().to_string_lossy()
        ));

        Ok(ProcessResult {
            status: "success".to_string(),
            output_file: Some(output),
            message: "图层合并完成".to_string(),
            details: json!({
                "source_layers_count": 5,
                "target_layer": target_layer
            }),
        })
    }
}
